src/main/kotlin/com/fantasy/ladbe/service/UserService.kt

```diff
@@ -4,32 +4,62 @@ import com.fantasy.ladbe.dto.UserDto
 import com.fantasy.ladbe.model.User
 import com.fantasy.ladbe.repository.UserRepository
 import org.springframework.stereotype.Service
-
+import org.springframework.transaction.annotation.Transactional
 
 @Service
 class UserService(val userRepository: UserRepository) {
 
-    fun create(
-        request: UserDto.Request.CreateUser,
-    ) {
-        val user = User(
-            kakaoCode = request.kakaoCode,
-            kakaoImagePath = request.kakaoImagePath,
-            kakaoEmail = request.kakaoEmail
-        )
-
-        userRepository.save(user)
+    /**
+     * 사용자의 정보를 업데이트 또는 회원가입을 해주는 메소드
+     * param :  kakaoCode - 사용자 고유 값
+     *          imagePath - 카카오 정보에 담겨져있는 프로필 사진의 경로
+     *          email - 카카오 정보에 담겨있는 사용자 이메일
+     * return : findOneByKakaoCode 메소드에서 반환값이 null이 아니라면, 업데이트를 한다.
+     *          null이 반환(사용자가 없음)이 된다면, 회원가입을 한다.
+     *          마지막으로 Dto로 변환하여 반환한다.
+     */
+    @Transactional
+    fun updateOrSave(
+        kakaoCode: Long,
+        imagePath: String,
+        email: String
+    ): UserDto.Response.UserDetail {
+        val user: User = findOneByKakaoCode(kakaoCode)?.let {
+            it.copy(
+                id = it.id,
+                kakaoCode = it.kakaoCode,
+                kakaoImagePath = imagePath,
+                kakaoEmail = email
+            )
+        }
+            ?: User(
+                kakaoCode = kakaoCode,
+                kakaoEmail = email,
+                kakaoImagePath = imagePath
+            )
+        return UserDto.Response.UserDetail()
+            .entityToDto(userRepository.save(user))
     }
 
+    /**
+     * 소셜에서 사용되는 고유 값을 이용해 한명의 이용자 찾기
+     * param :
+     *      kakaoCode - 사용자 고유 번호
+     * return : 이용자가 존재한다면, User Entity를 반환한다.
+     *          없다면, null을 반환한다.
+     */
+    private fun findOneByKakaoCode(kakaoCode: Long): User? {
+        return userRepository.selectByKakaoCode(kakaoCode)
+    }
```

Comment:
```suggestion
        val user: User = userRepository.selectByKakaoCode(kakaoCode)?.let {
            it.copy(
                id = it.id,
                kakaoCode = it.kakaoCode,
                kakaoImagePath = imagePath,
                kakaoEmail = email
            )
        }
            ?: User(
                kakaoCode = kakaoCode,
                kakaoEmail = email,
                kakaoImagePath = imagePath
            )
        return UserDto.Response.UserDetail()
            .entityToDto(userRepository.save(user))
    }
```

findOneByKakaoCode()의 사용이 updateOrSave에서만 있는 것이라면 하나로 합치는 것이 간결할 것 같습니다. (차후 findOneByKakaoCode()가 쓰이는 곳이 생길시 메소드 분리 해도 될 것 같아요)